compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5838,10 +5740,72 @@ public void visit(BLangGroupExpr groupExpr) {
     @Override
     public void visit(BLangRecordLiteral recordLiteral) {
         List<RecordLiteralNode.RecordField> fields =  recordLiteral.fields;
+        updateFieldsOfRecordLiteral(recordLiteral, fields);
         fields.sort((v1, v2) -> Boolean.compare(isComputedKey(v1), isComputedKey(v2)));
         result = rewriteExpr(rewriteMappingConstructor(recordLiteral));
     }
 
+    public void visit(BFunctionPointerInvocation functionPointerInvocation) {
+        result = functionPointerInvocation;
+    }
+
+    private List<String> getNamesOfRecordFields(List<RecordLiteralNode.RecordField> fields) {
+        List<String> fieldNames = new ArrayList<>();
+        for (RecordLiteralNode.RecordField field : fields) {
+            if (field.getKind() == NodeKind.RECORD_LITERAL_KEY_VALUE) {
+                BLangExpression key = ((BLangRecordLiteral.BLangRecordKeyValueField) field).key.expr;
+                if (key.getKind() == NodeKind.LITERAL) {
+                    fieldNames.add(((BLangLiteral) key).originalValue);
+                } else if (key.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+                    fieldNames.add(((BLangSimpleVarRef) key).variableName.value);
+                }
+            }
+        }
+        return fieldNames;
+    }
+
+    private void updateFieldsOfRecordLiteral(List<RecordLiteralNode.RecordField> fields, List<String> fieldNames,
+                                             Map<String, BInvokableSymbol> defaultValues, Location pos) {
+        for (Map.Entry<String, BInvokableSymbol> entry : defaultValues.entrySet()) {
+            String fieldName = entry.getKey();
+            if (fieldNames.contains(fieldName)) {
+                continue;
+            }
+            fieldNames.add(fieldName);
+            BInvokableSymbol invokableSymbol = entry.getValue();
+            BLangInvocation closureInvocation = getInvocation(invokableSymbol);
+            BLangRecordLiteral.BLangRecordKeyValueField member = new BLangRecordLiteral.BLangRecordKeyValueField();
+            member.key = new BLangRecordLiteral.BLangRecordKey(ASTBuilderUtil.createLiteral(pos, symTable.stringType,
+                                                                                                            fieldName));
+            member.valueExpr = addConversionExprIfRequired(closureInvocation, closureInvocation.getBType());
+
+            fields.add(member);
+        }
+    }
+
+    private void updateFieldsOfRecordLiteral(BLangRecordLiteral recordLiteral,
+                                             List<RecordLiteralNode.RecordField> fields) {
```

Comment:
Aren't these the user-specified fields? I would use

```suggestion
                                             List<RecordLiteralNode.RecordField> userSpecifiedFields) {
```

Reply:
Let's update wherever applicable.